### ISSUE TITLE
Make JSON settings more configurable

### ISSUE THREAD

**Issue by joshfriend** [flask-restful/flask-restful#373]
e.g:

```python
def output_json(data, code, headers=None):
    """Makes a Flask response with a JSON encoded body"""
    indent = current_app.config.get('JSON_INDENT', None)
    sort_keys = current_app.config.get('JSON_SORT_KEYS', False)

    dumped = json.dumps(data, indent=indent, sort_keys=sort_keys)

    resp = make_response(dumped, code)
    resp.headers.extend(headers or {})
    return resp
```

**Comment by Trii** [comment]
I was just looking at this as well. How about replacing `json.dumps` with `flask.json.dumps`? That way you can use a custom json encoder for the entire application the same way you can at this [snippet](http://flask.pocoo.org/snippets/119/)

```python
#!/usr/bin/env python
# -*- coding: utf-8 -*-
from flask import Flask, jsonify
from flask.json import JSONEncoder
import calendar
from datetime import datetime


class CustomJSONEncoder(JSONEncoder):

    def default(self, obj):
        try:
            if isinstance(obj, datetime):
                if obj.utcoffset() is not None:
                    obj = obj - obj.utcoffset()
                millis = int(
                    calendar.timegm(obj.timetuple()) * 1000 +
                    obj.microsecond / 1000
                )
                return millis
            iterable = iter(obj)
        except TypeError:
            pass
        else:
            return list(iterable)
        return JSONEncoder.default(self, obj)

app = Flask(__name__)
app.json_encoder = CustomJSONEncoder


@app.route('/custom')
def custom_jsonencoder():
    now = datetime.now()
    return jsonify({'now': now})

if __name__ == '__main__':
    app.run(debug=True)
```

**Comment by justanr** [comment]
Rather than simply having a bunch of configuration settings on the app object, why not centralize them into a dictionary stored on in `Flask.config`?

See: [snippet](https://github.com/justanr/flask-restful/blob/json_config/flask_restful/representations/json.py), #292

**Comment by joshfriend** [comment]
Yeah I think I do like that idea. It's simple and is more flexible than the one I proposed since you could store any valid argument to `json.dumps` in the settings dict.

**Comment by Trii** [comment]
Why does this plugin use different json encoding than flask does with [flask.json.jsonify](http://flask.pocoo.org/docs/0.10/api/#flask.json.jsonify) or [flask.json.dumps](http://flask.pocoo.org/docs/0.10/api/#flask.json.dumps)? These already have some application-level configuration options and would be more compatible and maintainable going forward.

**Comment by justanr** [comment]
@Trii That's actually a really good point. `flask_restful.representations.json.output_json` can tap into `flask.json.dumps` but just redirecting `from json import dumps` to `from flask.json import dumps` and mimicking `flask.json.jsonify` a little bit.

**Comment by joshfriend** [comment]
@Trii see #224

**Comment by Trii** [comment]
@joshfriend eh, well that sucks. There are a million reasons why using the flask core version is a better idea but I guess this is not the place.

**Comment by joshfriend** [comment]
fixed with #458
